flink-streaming-java/src/main/java/org/apache/flink/streaming/api/CheckpointingMode.java

```diff
@@ -74,5 +74,7 @@
 	 * scenarios, where a sustained very-low latency (such as few milliseconds) is needed,
 	 * and where occasional duplicate messages (on recovery) do not matter.</p>
 	 */
-	AT_LEAST_ONCE
+	AT_LEAST_ONCE,
+
+	UNALIGNED
```

Comment:
`UNALIGNED_AT_LEAST_ONCE`?

Reply:
Whether at_least or exactly for unaligned is up to how we implement the recovery process, but I guess it is probably supporting at-least mode if we really do it in formal way.

Considering the PoC not for failure recovery process, and we only implement the unaligned process ATM, so I think it also makes sense to only define `UNALIGNED` here.